apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ExecuteHelper.cs

```diff
@@ -14,14 +14,18 @@
 
 using Grpc.Core;
 using System;
-using System.Linq;
 using System.Threading.Tasks;
-using Google.Api.Gax;
 
 namespace Google.Cloud.Spanner.V1
 {
-    internal static class ExecuteHelper
+    /// <summary>
+    /// Helper class to wrap operations.
+    /// </summary>
+    public static class ExecuteHelper
     {
+        /// <summary>
+        /// Waits for <paramref name="task"/> to complete, handling session expiry by marking the session appropriately.
+        /// </summary>
         public static async Task<T> WithSessionChecking<T>(this Task<T> task, Func<Session> sessionFunc)
```

Comment:
Would be nice if this wasnt in the public api.

I think ideally, we'd be intercepting all the rpc calls and automatically tracking expired sessions that came from the pool.  Then the developer wouldnt have to do anything to have his expired sessions tracked properly (and excluded from the pool).

does vkit have any notion of an exception filter hook?